src/icTrento3d.cpp

```diff
@@ -0,0 +1,259 @@
+#include <cfloat>
+#include <cstdlib>
+#include <fstream>
+#include <iomanip>
+#include <iostream>
+#include <sstream>
+#include <cstring>
+#include "eos.h"
+#include "eoChiral.h"
+#include "fld.h"
+#include "icTrento3d.h"
+#include "rmn.h"
+#include "s95p.h"
+
+using namespace std;
+
+IcTrento3d::IcTrento3d(Fluid* f, const char* filename, double _tau0, const char* setup) {
+ cout << "loading TRENTO3d IC\n";
+ nx = f->getNX();
+ ny = f->getNY();
+ nz = f->getNZ();
+ dx = f->getDx();
+ dy = f->getDy();
+ dz = f->getDz();
+ xmin = f->getX(0);
+ xmax = f->getX(nx - 1);
+ ymin = f->getY(0);
+ ymax = f->getY(ny - 1);
+ zmin = f->getZ(0);
+ zmax = f->getZ(nz - 1);
+
+ tau0 = _tau0;
+
+ rho = new double**[nx];
+ for (int ix = 0; ix < nx; ix++) {
+  rho[ix] = new double*[ny];
+  for (int iy = 0; iy < ny; iy++) {
+   rho[ix][iy] = new double[nz];
+   for (int iz = 0; iz < nz; iz++) {
+    rho[ix][iy][iz] = 0.0;
+   }
+  }
+ }
+
+ // ---- read the events
+ nevents = 0;
```

Comment:
from what I see in icTrento3d.cpp, the module is not supposed to read more than one initial state table from Trento3d (there is no loop over nevents, and the locations of fdynamic array creation-destruction blocks do not assume more than one event either). If it is the case, could you please remove nevents variable from the module? This will make the structure less confusing.